document_similarity/microservice/routes/service.py

```diff
@@ -161,35 +142,59 @@ def update_similarities():
 
 @bp.route('/get_similarities', methods=['GET', 'POST'])
 def get_similarities():
-    # TODO: write documentation
+    # Retrieve query parameters
+    try:
 
-    # Retrieve query parameters 'document_id' and 'get_k'
-    if request.method == 'GET':
-        doc_id = request.args.get('document_id', default=None, type=int)
-        k = request.args.get('get_k', default=5, type=int)
-    elif request.method == 'POST':
-        doc_id = request.json['document_id']
-        k = request.json['get_k']
-    else:
-        # TODO: log exception
-        return abort(405)
+        # Retrieving parameters for method GET:
+        if request.method == 'GET':
+            doc_id = request.args.get('document_id', default=None, type=int)
+            k = request.args.get('limit', default=5, type=int)
+            page = request.args.get('page', default=None, type=int)
+            offset = request.args.get('offset', default=None, type=int)
+
+        # Retrieving parameters for method POST:
+        elif request.method == 'POST':
+            doc_id = request.json['document_id']
+            k = request.json['limit']
+            page = request.json['page']
+            offset = request.json['offset']
```

Comment:
I dont think we particularly need to allow both methods to be possible. It is completely fine to only allow 'GET' methods at this endpoint.

Reply:
Also, since we are following the REST API scheme:
- `GET` methods are used to retrieve the data from the service, while
- `POST` methods are used to send new data to the service to be stored.

I suggest removing the `POST` request since it does not follow the REST API scheme.

Reply:
If I understand correctly, the `POST` request should also be removed from the `update_similarities` method, since the only parameters given are the ID and language of the document?

Reply:
In general, we should stick to the following procedure:
* `GET` method is used, when we are only viewing data
* `POST` method is used, when our request will make changes to the database on the server.

Since at **update_similarities** we change the similarities tables with the document identified with document_id, it makes sense to allow only `POST` method on this request (and remove `GET` method).